From a23e8c3266677baa1ffaee0b775531642ebac622 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Tue, 22 Feb 2022 04:24:15 +0100
Subject: [PATCH] Add missing libdl link library for dlopen

glue/dl.cpp may call dlopen(), so Coin should be linked to the correct library,
which is available from CMAKE_DL_LIBS (and may be empty, e.g. on
glibc >= 2.34 or on FreeBSD, where dlopen is part of libc).

Fixes #473.
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f32f6ff..29f03e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -530,6 +530,7 @@ else()
   endif()
   check_library_exists(dl dlopen "" HAVE_DL_LIB)
   if(HAVE_DL_LIB)
+    list(APPEND COIN_TARGET_LINK_LIBRARIES ${CMAKE_DL_LIBS})
     check_include_file(dlfcn.h HAVE_DLFCN_H)
   endif()
   check_library_exists(dld shl_load "" HAVE_DLD_LIB)
-- 
2.35.1

